mitol-django-mail/testapp/settings.py

```diff
@@ -0,0 +1,129 @@
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "0541x5@jg^ulh@1gk&9swit)b_@z5by#q*-6ofig@rv!al#1)="
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
+ALLOWED_HOSTS = []
+
+
+# Application definition
+
+INSTALLED_APPS = [
+    "django.contrib.admin",
+    "django.contrib.auth",
+    "django.contrib.contenttypes",
+    "django.contrib.sessions",
+    "django.contrib.messages",
+    "django.contrib.staticfiles",
+    # test the reusabe app
+    "mitol.mail.apps.MailApp",
+]
+
+MIDDLEWARE = [
+    "django.middleware.security.SecurityMiddleware",
+    "django.contrib.sessions.middleware.SessionMiddleware",
+    "django.middleware.common.CommonMiddleware",
+    "django.middleware.csrf.CsrfViewMiddleware",
+    "django.contrib.auth.middleware.AuthenticationMiddleware",
+    "django.contrib.messages.middleware.MessageMiddleware",
+    "django.middleware.clickjacking.XFrameOptionsMiddleware",
+]
+
+ROOT_URLCONF = "testapp.urls"
+
+TEMPLATES = [
+    {
+        "BACKEND": "django.template.backends.django.DjangoTemplates",
+        "DIRS": [BASE_DIR + "/testapp/templates/"],
+        "APP_DIRS": True,
+        "OPTIONS": {
+            "context_processors": [
+                "django.template.context_processors.debug",
+                "django.template.context_processors.request",
+                "django.contrib.auth.context_processors.auth",
+                "django.contrib.messages.context_processors.messages",
+            ],
+        },
+    },
+]
+
+WSGI_APPLICATION = "testapp.wsgi.application"
+
+
+# Database
+# https://docs.djangoproject.com/en/2.2/ref/settings/#databases
+
+DATABASES = {
+    "default": {
+        "ENGINE": "django.db.backends.sqlite3",
+        "NAME": os.path.join(BASE_DIR, "db.sqlite3"),
+    }
+}
+
+
+# Password validation
+# https://docs.djangoproject.com/en/2.2/ref/settings/#auth-password-validators
+
+AUTH_PASSWORD_VALIDATORS = [
+    {
+        "NAME": "django.contrib.auth.password_validation.UserAttributeSimilarityValidator",
+    },
+    {"NAME": "django.contrib.auth.password_validation.MinimumLengthValidator",},
+    {"NAME": "django.contrib.auth.password_validation.CommonPasswordValidator",},
+    {"NAME": "django.contrib.auth.password_validation.NumericPasswordValidator",},
+]
+
+
+# Internationalization
+# https://docs.djangoproject.com/en/2.2/topics/i18n/
+
+LANGUAGE_CODE = "en-us"
+
+TIME_ZONE = "UTC"
+
+USE_I18N = True
+
+USE_L10N = True
+
+USE_TZ = True
+
+
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/2.2/howto/static-files/
+
+STATIC_URL = "/static/"
+
+
+SITE_NAME = "Test app"
+SITE_BASE_URL = "http://127.0.0.1:8000/"
+
+
+# mail app settings
+MITOL_MAIL_MESSAGE_CLASSES = ["testapp.messages.SampleMessage"]
+
+MITOL_MAIL_FROM_EMAIL = "invalid@localhost"
+MITOL_MAIL_REPLY_TO_ADDRESS = "invalid@localhost"
+MITOL_MAIL_ENABLE_EMAIL_DEBUGGER = True
```

Comment:
In order to actually send an email in this test app, I needed to add `ANYMAIL_MAILGUN_API_KEY` and `MAILGUN_SENDER_DOMAIN` settings (and ran the code in a Django shell). I think we'll benefit from this test app being more readily usable. Some possibilities to that end:
- Allow the user to provide the above settings (and perhaps other settings like `MITOL_MAIL_FROM_EMAIL`, `MITOL_MAIL_REPLY_TO_ADDRESS`) as environment variables
- Add a short README to the `testapp` directory, which would include details about providing those settings, as well as a short code snippet to illustrate basic usage:
    ```python
    ### Create a user
    from django.contrib.auth import get_user_model
    User = get_user_model()
    user = User.objects.create(username="<my-username>", email="<my-email>")
    
    ### Send an email
    from mitol.mail.api import get_message_sender
    from mitol.mail.messages import TemplatedMessage
    from testapp.messages import SampleMessage
    from django.contrib.auth import get_user_model
    User = get_user_model()
    user = User.objects.get(username="<my-username>")
    with get_message_sender(SampleMessage, shared_context={}) as sender:
        sender.build_and_send_message(user, {})
    ```

Reply:
I added the readme for `testapp`.

I'm not sure I agree that environment variable parsing is a good thing inside a reusable app, it blurs the separation of concerns there quite a bit and I don't think I've run across any other reusable apps in the ecosystem that go that route. I totally agree though that it's some amount of boilerplate that we end up copy/pasting in various apps though, so it's definitely a problem worth solving. I think we should do it separate from this work since it needs design around how to accomplish it, particularly how to integrate things like the `generate_app_json` command in bootcamp/xpro.

I'll open an issue on here so we remember to circle back on it.

For now, since all of the apps currently using the `mail` app already configure things in their own `settings.py`, we can easily remap those variable to the new `settings.py` keys.